src/parser.js

```diff
@@ -727,6 +736,7 @@ export default class Parser extends Stream {
 
           // prepare for the next URI
           currentUri = {};
+          this.addPDTToSegment_();
```

Comment:
Is there a better way to call the `addPDTToSegment_` function for every segment that's parsed?